### DISCUSSION TITLE
target: <database>.-.primary: vttablet: rpc error: code = Aborted desc = transaction <transaction>: in use: in use: for tx killer rollback

### DISCUSSION THREAD

**Discussion by mnoquiao** [planetscale/discussion#416] [Bug Reports] [answered]
Hello! I have recently performed a migration of our database to PlanetScale. Currently, I am encountering an error in the Schedule Queue Job that runs on a daily basis. I am uncertain about the underlying cause of this issue.

Here is the error log I get from my application logs:
```
[2023-03-13 17:00:28] production.ERROR: SettleAgentsCommission Job: 2023-03-13T17:00:28
 - App\Jobs\SettleAgentsCommission::handle
 - Line: 760
 - SQLSTATE[HY000]: General error: 1105 target: casinoq-stg.-.primary: vttablet: rpc error: code = Aborted desc = transaction 1678643446311755190: in use: in use: for tx killer rollback (CallerID: dnquad7klknm7v2fkurr) (SQL: update `admins_assets` set `balance` = `balance` + 3.7800, `admins_assets`.`updated_at` = 2023-03-13 17:00:28 where `admin_id` = 32)
```  

Here is the error I got from PlanetScale dashboard:
```
[target: casinoq-stg.-.primary: vttablet: rpc error: code = Aborted desc = transaction <transaction>: in use: in use: for tx killer rollback](https://app.planetscale.com/mark-aiowireless/casinoq-stg/main/insights/errors/81b718df847d70136d81425f6cbaef256867e928de2c6f30ed9877bc1731dc71?)
```

And more detailed below:
2023-03-14 01:00:28
```
Ran in 1 ms
[update admins_assets set balance = balance + :vtg1, admins_assets.updated_at = :v1 where admin_id = :v2](https://app.planetscale.com/<organization-name>/<database>/main/insights/63f0bda849155bb42b0338290374a8aa34dc5207dbec455813198387408da9c5?keyspace=casinoq-stg)
```

**Comment by fooforge** [comment, marked as answer]
Hi @mnoquiao,

This is due to our 20 seconds hard timeout for transactions. We have more information on this and other limits [in our documentation](https://planetscale.com/docs/reference/planetscale-system-limits) and I have also [blogged about it](https://planetscale.com/blog/supports-notes-from-the-field) a few weeks ago.

I suspect that, while the query's been executed quickly, your code's kept the transaction open for longer than necessary which has then resulted in the timeout to be hit. You will need to optimize your workloads and better control how long your application keeps the connection open to not hit these limits.

**Reply by mnoquiao** [reply]
Thank you for your help, I reviewed the docs and make the necessary changes to our application code.
